Automattic/woocommerce-payments
# Only disable nonce checks when in dev mode

#### Changes proposed in this Pull Request

Makes it so that the nonce check is for Store API requests is only disabled when 2 conditions are fulfilled:

1. The store is eligible for platform checkout functionality; and
2. The store is in dev mode.









#### Testing instructions





Easiest to test with a debugger.

1. Set a breakpoint [here](https://github.com/Automattic/woocommerce-payments/blob/a8da33d4cf99c0cccaf5dfab43254787122dbcfd/includes/class-wc-payments.php#L889).
2. Set your store to dev mode.
3. Reload the front page of your store.
4. After you hit the breakpoint, step through the code to make sure the nonce check is disabled.
5. Now disable dev mode.
6. Reload the front page of your store.
7. After you hit the breakpoint, step through the code to make sure the nonce check **is not** disabled.

Also:

Go through the whole payment flow:

1. Add a product to the cart on the merchant store.
2. Go the **Checkout** page and start a WooPay session.
3. Change the number of items in your cart **from WooPay**.
4. Make a payment and make sure it's completed.

-------------------

    - No changelog entry because this functionality is not generally available yet.

**Post merge**



    - No testing instructions; nothing to test at the moment.
